### ISSUE TITLE
Rechart breaks in Nextjs 15-rc.0

### ISSUE THREAD

**Issue by john-prutton** [vercel/next.js#67052]
### Link to the code that reproduces this issue

https://github.com/john-prutton/recharts-nextjs-15

### To Reproduce

1. clone, pnpm install, and start app. the latest commit uses nextjs 15
1.1 go to http://localhost:3000/ and see that chart is not visible

2. rollback one commit with something like `git reset head~1 --hard`. this commit uses next 14
2.1 pnpm install packages and restart app.
2.2 go to http://localhost:3000 and see that the the graph shows up and is functioning as expected

### Current vs. Expected behavior

Here the graph is working in next 14:
![next-14-working](https://github.com/vercel/next.js/assets/19376890/6d4241ac-c45c-402a-b380-d567b1e4befc)

Here the graph is not working in next 15:
![next-15-broken](https://github.com/vercel/next.js/assets/19376890/742aea1e-ee78-4e47-90fd-9078328ba410)

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Pro
  Available memory (MB): 16315
  Available CPU cores: 16
Binaries:
  Node: 20.11.1 (no difference on 22.3.0)
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 15.0.0-rc.0 // Latest available version is detected (15.0.0-rc.0). 
  eslint-config-next: 14.2.4
  react: 19.0.0-rc-f994737d14-20240522
  react-dom: 19.0.0-rc-f994737d14-20240522
  typescript: 5.4.5
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Which stage(s) are affected? (Select all that apply)

next dev (local), next start (local)

### Additional context

Working in  "next": "14.2.4", broken in "next": "15.0.0-rc.0"

**Comment by Netail** [comment]
Maybe more related to rechart with React 19, than NextJS specific (https://github.com/recharts/recharts/issues/4558)

**Comment by nshout** [comment]
Indeed, it is related to React 19. Recharts [2.13.0-alpha.4](https://www.npmjs.com/package/recharts/v/2.13.0-alpha.4) still mostly does not work with React 19. You'll have to hang on a bit or downgrade NextJS to v14 with React v18.

**Comment by tobiaslins** [comment]
@nshout don't forget to rewrite `react-is` https://github.com/recharts/recharts/issues/4558#issuecomment-2132375528

**Comment by moog16** [comment]
I had a really difficult time getting it to work w/ R19. Almost to the point where I was going to downgrade back to 18+. This not is here for anyone who needs something else to try for debugging / getting it to work:

1. delete all overrides from package.json
2. delete package-lock.json
3. delete `recharts`, `react`, and `react-dom` dep from package.json
4. `npm i react@latest react-dom@latest`
5. add back all overrides from step 1
6. `npm i recharts@latest`

**Comment by moog16** [comment]
I had a really difficult time getting it to work w/ R19. Almost to the point where I was going to downgrade back to 18+. This not is here for anyone who needs something else to try for debugging / getting it to work:

1. delete all overrides from package.json
2. delete package-lock.json
3. delete `recharts`, `react`, and `react-dom` dep from package.json
4. `npm i react@rc react-dom@rc`
5. add back all overrides from step 1
6. `npm i recharts@latest --force`
7. add an override for recharts: `"recharts": { "react": "$react", "react-dom": "$react-dom" }`
8. `npm i`

Hopefully that works or at least inpsires something else to try.

**Comment by abdulsalamcodes** [comment]
thank you @moog16 , this solution worked for me.

**Comment by marcoshernanz** [comment]
Thanks for the report. I can reproduce this with the original Recharts 2.12.7 / React 19 RC setup, but it comes from Recharts' incompatible `react-is` dependency rather than Next.js. Recharts tracked this in [recharts/recharts#4558](https://github.com/recharts/recharts/issues/4558) and fixed the dependency model in v3. I verified the same chart renders on current Next.js 15 and 16 with Recharts 3, so I'm closing this as resolved upstream.
